engine/servermaster/campaign.go

```diff
@@ -198,7 +198,7 @@ func (s *Server) watchLeader(ctx context.Context, m *rpcutil.Member, key []byte,
 	m.IsLeader = true
 	s.leader.Store(m)
 
-	s.createLeaderClient(ctx, m.AdvertiseAddr)
+	s.createLeaderClient(m.AdvertiseLeaderAddr)
```

Comment:
Since the `advertise-leader-addr` will receive forward RPC, `advertise-addr` will receive executor RPC, should the server binds to `0.0.0.0:<port>`, and both `advertise-leader-addr` and `advertise-addr` use this port in their address.

Reply:
`advertise(-leader)-addr` port is not required to be the same as the bind port. The `advertise(-leader)-addr` is the address that advertises the client to connect, it may be a service endpoint or any normal L4 or L7 proxy address.